src/test/java/com/tuplejump/stargate/util/Record.java

```diff
@@ -0,0 +1,103 @@
+package com.tuplejump.stargate.util;
+
+import com.datastax.driver.core.Row;
+
+import java.lang.reflect.Array;
+import java.util.*;
+
+public class Record {
+
+    private Map recordDefinition = new HashMap<String, String>();
+    private Map<String, Object> record = new HashMap<String, Object>();
+
+    public Record(String[] fields, Object[] values, String[] type) {
+        if (fields.length == values.length) {
+            for (int i = 0; i < fields.length; i++) {
+                recordDefinition.put(fields[i].toLowerCase(), type[i]);
+                record.put(fields[i].toLowerCase(), values[i]);
+            }
+        }
+    }
+
+    Record(Map record) {
+        this.record = record;
```

Comment:
Can you add a null check here? So that we can avoid having to check for the HashMap != null everywhere else